### ISSUE TITLE
Docs: differences between recommended and strict configs not mentioned on the website

### ISSUE THREAD

**Issue by doberkofler** [typescript-eslint/typescript-eslint#8712]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

The playground does currently not allow to specify 7.3.0

### Repro Code

The rule `@typescript-eslint/no-floating-promises` no longer seems to have the option `ignoreVoid` set to true by default in 7.3.0.
When upgrading from 7.2.0 to 7.3.0 the following code report an error and only explicitly specifying `ignoreVoid: true` it works as in 7.2.0.

```TypeScript
const get = async (): Promise<void> => {};

export const useGet = (): void => {
	void get();
};
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-floating-promises": ["error"],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

No errors when using void

### Actual Result

Even when using void an error is reported

**Comment by JoshuaKGoldberg** [comment]
I can't reproduce this in the playground locally. https://typescript-eslint.io/play/#ts=5.4.2&fileType=.ts&code=IYZwngdgxgBAZgV2gFwJYHsIwO7oE4DWAFAJQwDeMAvgFA0Bu6qAJjvsSQNw1A&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oDN4OBDfMwDmtYtA4BbSshTooiaBOiRwYAL4h1QA&tsconfig=&tokens=false will be the link once #8714 is merged & fixes the public website deploy (sorry about that).

The default config for the rule hasn't changed:

https://github.com/typescript-eslint/typescript-eslint/blob/7c77b1b123d13fb585d588a9dd19f810c973a42a/packages/eslint-plugin/src/configs/recommended-type-checked.ts#L23

#8364 added the option for rules to have different defaults in `recommended` vs `strict`. Is that what you're seeing, maybe?

**Comment by JoshuaKGoldberg** [comment]
Got it - I'm realizing now I really should have done more to document this. Sorry for the confusion. I'll retarget this as a docs issue and send a PR very soon.

**Comment by bradzacher** [comment]
@JoshuaKGoldberg i would probably say that this is one of those changes a lot of people would disagree with and want to reconfigure.

Using `void` to demark a promise as intentionally floating is something a lot of people like as it's much less verbose than a disable comment (and doesn't have the same stigma).

The other config changes are tweaks to rules - but this one is a pretty big shift in the coding style.

Perhaps we should revert this specific config change and save it for a major, given how disruptive it will be for users?

**Comment by JoshuaKGoldberg** [comment]
Yeah, makes sense. I'll send that to. 👍

**Comment by JoshuaKGoldberg** [comment]
Good idea, thanks! Added to https://github.com/typescript-eslint/typescript-eslint/releases/tag/v7.3.0.

**Comment by JoshuaKGoldberg** [comment]
https://github.com/typescript-eslint/typescript-eslint/releases/tag/v7.3.1 is released with the revert.
